gtest_parallel_tests.py

```diff
@@ -159,35 +148,106 @@ def test_run_task_basic(self):
 
     self.assertEqual(task_manager.global_exit_code, 1)
 
-  @unittest.skipIf(sys.platform == 'win32', 'non-Windows specific')
-  def test_get_save_file_path_unix(self):
-    self.assertEqual(os.path.join(os.path.expanduser('~'),
-                                  '.cache', 'gtest-parallel'),
-                     gtest_parallel.get_save_file_path())
 
-    with guard_environ('XDG_CACHE_HOME', os.getcwd()):
-      self.assertEqual(os.path.join(os.getcwd(), 'gtest-parallel'),
-                       gtest_parallel.get_save_file_path())
+@contextlib.contextmanager
+def guard_environ(var, val):
```

Comment:
Is this something we can mock out as os.environ = new_environ, and finally: os.environ = old_environ? Instead of replacing values in the environment I'd rather we replaced it (so nothing in the tests can accidentally rely on the test environment).